src/dbnode/storage/index/block.go

```diff
@@ -603,6 +576,47 @@ func (b *block) queryWithSpan(
 	return nil
 }
 
+func (b *block) docWithinQueryRange(doc doc.Document, opts QueryOptions) bool {
+	md, ok := doc.Metadata()
+	if !ok || md.OnIndexSeries == nil {
+		return true
+	}
+
+	onIndexSeries, cleanup, reconciled := md.OnIndexSeries.ReconciledOnIndexSeries()
+	if reconciled {
+		b.metrics.reconciledEntryOnQuery.Inc(1)
+	}
```

Comment:
nit: Maybe increment a metric when it's not reconciled too? That way we can tell the magnitude of entries that are reconciled vs not-reconciled (i.e. is 1k/sec big or small number? hard to know if we don't know the total).